private/buf/cmd/buf/command/alpha/repo/reposync/sync_handler.go

```diff
@@ -88,30 +87,22 @@ func (h *syncHandler) ResolveSyncPoint(ctx context.Context, module bufmoduleref.
 	return hash, nil
 }
 
-func (h *syncHandler) CheckSyncedGitCommits(ctx context.Context, module bufmoduleref.ModuleIdentity, commitHashes map[string]struct{}) (map[string]struct{}, error) {
+func (h *syncHandler) IsGitCommitSynced(ctx context.Context, module bufmoduleref.ModuleIdentity, hash git.Hash) (bool, error) {
 	service := connectclient.Make(h.clientConfig, module.Remote(), registryv1alpha1connect.NewLabelServiceClient)
 	res, err := service.GetLabelsInNamespace(ctx, connect.NewRequest(&registryv1alpha1.GetLabelsInNamespaceRequest{
 		RepositoryOwner: module.Owner(),
 		RepositoryName:  module.Repository(),
 		LabelNamespace:  registryv1alpha1.LabelNamespace_LABEL_NAMESPACE_GIT_COMMIT,
-		LabelNames:      stringutil.MapToSlice(commitHashes),
+		LabelNames:      []string{hash.Hex()},
 	}))
 	if err != nil {
 		if connect.CodeOf(err) == connect.CodeNotFound {
 			// Repo is not created
-			return nil, nil
-		}
-		return nil, fmt.Errorf("get labels in namespace: %w", err)
-	}
-	syncedHashes := make(map[string]struct{})
-	for _, label := range res.Msg.Labels {
-		syncedHash := label.LabelName.Name
-		if _, expected := commitHashes[syncedHash]; !expected {
-			return nil, fmt.Errorf("received unexpected synced hash %q, expected %v", syncedHash, commitHashes)
+			return false, nil
 		}
-		syncedHashes[syncedHash] = struct{}{}
+		return false, fmt.Errorf("get labels in namespace: %w", err)
 	}
-	return syncedHashes, nil
+	return len(res.Msg.Labels) == 1, nil
```

Comment:
May be make sure that the returned label is actually the one requested?

Reply:
I thought about this, but this is going away soon anyway, to be replaced by `GetReferenceByName{ Ref: git.Hash() }`, so I didn't bother.